### PR TITLE
Complete ride

### LINKED ISSUE #105: Complete rides
* Feature name: Closable rides
* Description: It should be possible to close rides once they are complete.
* Requires: Creating rides
* Time Estimation: 45 minutes

### PR DESCRIPTION
**Pull request by jpark052**
I have added "Active rides" and "Previous rides" tabs, and when a user clicks "complete ride" button in active ride it's marked as complete and moves to "Previous rides" tab.

Couple of questions:
- Do we want "un-complete" button that makes a completed ride active again as well?
- Who should be able to "complete" the ride? anyone in the group? or just the driver?

resolves #105

### CODE REVIEW

**Comment by CobeyH** [comment]
I think that we shouldn't have an "un-complete ride" but maybe a check to make sure that the start date of the ride has passed before the ride is closed? Also, should rides be closed if all participants leave?

**Comment by jpark052** [comment]
> 

Sounds great, I can add the start date checking once #145 is merged.
Also I think it makes sense to close the ride if all participants leave. I will implement that function as well.

**Comment by CobeyH** [comment]
If you want you can finish and merge #145 . I was saving that issue for Sam because it should be reasonable easy to finish. However, she isn't around until Wednesday so we can give her #84 instead.

File: frontend/src/components/RideCard.tsx [hunk 1]
```diff
@@ -249,6 +264,31 @@ function DriverDisplay({
...
+  userId,
+  driverId,
+}: {
+  rideId: string;
+  userId: string;
+  driverId: string | undefined;
+}) {
+  return (
+    <Button
+      disabled={driverId == undefined || userId !== driverId}
```
Comment (CobeyH) [inline frontend/src/components/RideCard.tsx:278]: Could we not render this button at all if they aren't the driver? I feel that having a button that they can never use may just be confusing. If you want to show the button then maybe a tooltip saying that the driver is the only one that can complete the ride would be good?

**Review by CobeyH — APPROVED** [review]
I left a minor comment but I see no blocking issues. Feel free to change the button behaviour if you want to and then merge this.

File: frontend/src/components/RideCard.tsx (hunk 1 above)
Comment (jpark052) [reply frontend/src/components/RideCard.tsx:278]: Yeah that makes sense. Now only the driver can see the "complete" button

**Pull request merged**